tests/d3d12_mesh_shader.c

```diff
@@ -0,0 +1,2326 @@
+        0x09, 0x25, 0x2b, 0x09, 0x28, 0x8d, 0x00, 0x8c, 0x11, 0x80, 0x20, 0x08, 0xe2, 0xdf, 0x18, 0x01, 0x08, 0x82, 0x20, 0x08, 0x06, 0x63, 0x04, 0x20, 0x08, 0x82, 0xf8, 0x2f, 0x8c, 0x11, 0x80, 0x20,
+        0x08, 0xa2, 0x60, 0x00, 0x00, 0x23, 0x06, 0x05, 0x00, 0x82, 0x60, 0x60, 0x88, 0x41, 0x32, 0x62, 0x70, 0x00, 0x20, 0x08, 0x06, 0x8e, 0x97, 0x64, 0x5d, 0x05, 0x9d, 0x5e, 0x90, 0xca, 0x02, 0x43,
+        0x3e, 0x45, 0x68, 0x7a, 0x41, 0x2a, 0x23, 0x0e, 0xf9, 0x98, 0xa0, 0xc8, 0xc7, 0x04, 0x05, 0x3e, 0x26, 0x2c, 0xf0, 0x19, 0x31, 0x50, 0x00, 0x10, 0x04, 0x03, 0x65, 0x0d, 0xaa, 0x31, 0x18, 0x03,
+        0x4d, 0x50, 0x46, 0x0c, 0x14, 0x00, 0x04, 0xc1, 0x40, 0x59, 0x83, 0x6a, 0x0c, 0xc6, 0xe0, 0x0a, 0x94, 0x11, 0x03, 0x05, 0x00, 0x41, 0x30, 0x50, 0xd6, 0xa0, 0x1a, 0x83, 0x31, 0xc8, 0x1e, 0x65,
+        0xc4, 0x40, 0x01, 0x40, 0x10, 0x0c, 0x94, 0x35, 0xa8, 0xc6, 0x60, 0x0c, 0x38, 0x47, 0x19, 0x31, 0x50, 0x00, 0x10, 0x04, 0x03, 0x65, 0x0d, 0x2a, 0x33, 0x18, 0x03, 0xed, 0x50, 0x46, 0x0c, 0x14,
+        0x00, 0x04, 0xc1, 0x40, 0x59, 0x83, 0xca, 0x0c, 0xc6, 0xe0, 0x2a, 0x94, 0x52, 0xcc, 0x00, 0x46, 0x0c, 0x14, 0x00, 0x04, 0xc1, 0x80, 0x59, 0x03, 0x2b, 0x0c, 0xc8, 0x60, 0x0b, 0x96, 0xe1, 0x86,
+        0x85, 0x0c, 0xc0, 0x60, 0x96, 0x21, 0x10, 0x82, 0x11, 0x83, 0x04, 0x00, 0x41, 0x30, 0x20, 0xde, 0x60, 0x2a, 0x83, 0x32, 0x40, 0x03, 0x31, 0x18, 0x31, 0x50, 0x00, 0x10, 0x04, 0x03, 0x65, 0x0d,
+        0xac, 0x32, 0x28, 0x03, 0x2e, 0x2a, 0x83, 0x11, 0x03, 0x05, 0x00, 0x41, 0x30, 0x50, 0xd6, 0xc0, 0x2a, 0x83, 0x32, 0xc8, 0xa0, 0x32, 0x18, 0x31, 0x50, 0x00, 0x10, 0x04, 0x03, 0x65, 0x0d, 0xac,
+        0x32, 0x28, 0x83, 0x2d, 0x2a, 0x83, 0x11, 0x03, 0x05, 0x00, 0x41, 0x30, 0x50, 0xd6, 0xc0, 0x2a, 0x83, 0x32, 0xf0, 0xa0, 0x32, 0x18, 0x31, 0x50, 0x00, 0x10, 0x04, 0x03, 0x46, 0x0d, 0x2c, 0x34,
+        0x28, 0x03, 0x4e, 0x2a, 0x83, 0x59, 0x02, 0x01, 0x01, 0x00, 0x00, 0x00, 0x00,
+    };
+    static const union d3d12_shader_bytecode_subobject ms_interface_matching_subobject =
+            {{ D3D12_PIPELINE_STATE_SUBOBJECT_TYPE_MS, { ms_interface_matching, sizeof(ms_interface_matching) } }};
+
+#if 0
+    StructuredBuffer<uint> buf : register(t0);
+
+    struct PrimData
+    {
+        uint prim_id : UV_PRIMITIVE_ID;
+    };
+
+    groupshared uint primitive_count;
+
+    [numthreads(32,1,1)]
+    [outputtopology("triangle")]
+    void main(in uint tid : SV_GroupIndex, in uint3 gid : SV_GroupID, in uint3 did : SV_DispatchThreadID,
+            out vertices float4 v[3] : SV_POSITION, out primitives PrimData p[32], out indices uint3 i[32]) {
+        if (tid == 0)
+            primitive_count = 0;
+
+        GroupMemoryBarrierWithGroupSync();
+
+        uint cull_mask = buf[gid.x];
+        uint primitive_index = ~0u;
+
+        if (cull_mask & (1u << tid))
+            InterlockedAdd(primitive_count, 1u, primitive_index);
+
+        GroupMemoryBarrierWithGroupSync();
+        SetMeshOutputCounts(3, primitive_count);
+
+        if (primitive_count == 0)
```

Comment:
It seems a bit spicy to emit 3 vertices + 0 primitives, but it should be legal, albeit bizarre.

Reply:
The AMD driver hangs if we don't call `SetMeshOutputCounts` prior to returning, but yeah, this works on both drivers now and is probably something we should end up supporting.